include/rcutils/filesystem.h

```diff
@@ -243,6 +244,75 @@ RCUTILS_PUBLIC
 size_t
 rcutils_get_file_size(const char * file_path);
 
+/// An iterator used for enumerating directory contents
+typedef struct rcutils_dir_iter_t
+{
+  /// The name of the enumerated file or directory
+  const char * entry_name;
```

Comment:
I should probably note that this value is only valid until the next call to `rcutils_dir_iter_next` or `rcutils_dir_iter_end`.

Reply:
I think that is fine; that is the same behavior as `readdir` in POSIX.